### ISSUE TITLE
location_permissions does not exist

### ISSUE THREAD

**Issue by andrewzakhartchouk** [rekab-app/background_locator#184]
When trying out the example app provided in the package on an Android virtual device, I found that it was not attempting to request location permissions. Manually setting the app's permission in the settings menu fixed the problem.

However checking the code I found that the package location_permissions is non-existent, and thus the related code cannot function.

![image](https://user-images.githubusercontent.com/65052923/104552062-65928780-567b-11eb-81b1-6779028469c5.png)
![image](https://user-images.githubusercontent.com/65052923/104552088-72af7680-567b-11eb-8393-3b8aa7cc5308.png)

Because there might be some additional setup for the example app that I didn't complete, I thought I would check here first before raising an issue at the location_permissions repository.

**Comment by mehdok** [comment]
HI @andrewzakhartchouk 
Thank you for opening an issue;

Did you update the `location permission` package?

**Comment by andrewzakhartchouk** [comment]
If by updating you mean `pub get`, yes I have. My location_permissions is version ^3.0.0+1.

**Comment by mehdok** [comment]
By updating, I mean increasing the version of the library;

> the request is still being denied and returns PermissionStatus.denied

I'll check this;

**Comment by mehdok** [comment]
I believe this is a `location_permissions ` bug and there is nothing we can do on the library side. All we need is to just get location permission, you are free to use whatever is working for this matter;

**Comment by andrewzakhartchouk** [comment]
Thanks for looking into it. Perhaps I misunderstood how the permissions work, because if the app permission is set to 'Denied' then the location_permissions request won't work. Only if it's set to 'Allowed Only While In Use' will it automatically redirect to the app's permission page.

This also applies to the flutter-permission-handler.

My workaround ends up being using the app_settings package to redirect.

**Comment by mehdok** [comment]
I don't think you misunderstood something, something definitely has changed, because the default permission status should not be `denied`, I believe it should be `not_determined` or something similar;

Anyway, I would be more than happy to hear again from you and your solution for this problem;

**Comment by andrewzakhartchouk** [comment]
I can make a pull request for the example app that demonstrates using the app_settings package if you would like.

**Comment by mehdok** [comment]
I would appreciate it if you do so;
